osbuild/testutil/__init__.py

```diff
@@ -155,3 +159,30 @@ def pull_oci_archive_container(archive_path, image_name):
         yield
     finally:
         subprocess.check_call(["skopeo", "delete", f"containers-storage:{image_name}"])
+
+
+def make_fake_service_fd() -> int:
+    """Create a file descriptor suitable as input for --service-fd for any
+    host.Service
+
+    Note that the service will take over the fd and take care of the
+    lifecycle so no need to close it.
+    """
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_SEQPACKET)
+    fd = os.dup(sock.fileno())
+    return fd
+
+
+def find_one_subclass_in_module(module: ModuleType, subclass: Type) -> object:
+    """Find the class in the given module that is a subclass of the given input
+
+    If multiple classes a found an error is raised.
+    """
+    cls = None
+    for memb in inspect.getmembers(
+            module,
+            predicate=lambda obj: inspect.isclass(obj) and issubclass(obj, subclass)):
+        if cls:
+            raise ValueError(f"already have {cls}, also found {memb}")
+        cls = memb[1]
```

Comment:
Why `memb[1]` not `memb[0]` ?

Reply:
I clarified this a bit better now. The `getmembers()` returns a tuple of (name, value) and we are only interested in the `value` here but the name is still useful for debugging so I use it now in the error case :)

Reply:
Thanks for the explanation:)